iothub/service/src/ServiceAuthenticationWithSharedAccessPolicyToken.cs

```diff
@@ -21,59 +21,70 @@ public sealed class ServiceAuthenticationWithSharedAccessPolicyToken : IAuthenti
         /// <param name="token">Token associated with the shared access policy.</param>
         public ServiceAuthenticationWithSharedAccessPolicyToken(string policyName, string token)
         {
-            this.SetPolicyName(policyName);
-            this.SetToken(token);
+            SetPolicyName(policyName);
+            SetToken(token);
         }
 
+        /// <summary>
+        /// The name of the policy
+        /// </summary>
         public string PolicyName
         {
-            get { return this.policyName; }
-            set { this.SetPolicyName(value); }
+            get => _policyName;
```

Comment:
same - this can be simplified to remove the backing field and have setter with null check

Reply:
ignore, auto properties cannot be used to perform validations